proposals/NNNN-improve-starting-streaming-services.md

```diff
@@ -0,0 +1,54 @@
+# Improve starting streaming services
+
+* Proposal: [NNNN](NNNN-improve-starting-streaming-services.md)
+* Author: [Kujtim Shala](https://github.com/smartdevicelink/kshala-ford)
+* Status: **Awaiting review**
+* Impacted Platforms: [ iOS / Java Suite / JavaScript Suite ]
+
+## Introduction
+
+In the process of analyzing mobile navigation streaming issues, we have identified several issues and unaligned implementations between SDL Core and the libraries. This proposal addresses difficulties to handle failed `StartService` attempts for video and audio services including
+1. synchronizing `StartService` requests
+2. reset SSL connection upon failures
+3. retry mechanism for starting services
+
+## Motivation
+
+With current SDL implementation, if the app sends `StartService` with `encrypted=1` and Core's `SSLMode` is set to `CLIENT`,  Core starts a TLS handshake to authenticate the app which is required to send encrypted data on the service session. If this service is successfully started, Core will remember the handshake for the current transport connection. If Core receives a second `StartService` request with `encrypted=1` it will not restart another TLS handshake. However, if the first StartService failed, Core will reset the SSL connection and restart a TLS handshake for the next `StartService` request.
```

Comment:
```suggestion
With current SDL implementation, if the app sends `StartService` with `encrypted=1` and Core's `SSLMode` is set to `CLIENT`, Core starts a TLS handshake to authenticate the app which is required to send encrypted data on the service session. If this service is successfully started, Core will remember the handshake for the current transport connection. If Core receives a second `StartService` request with `encrypted=1` it will not restart another TLS handshake. However, if the first `StartService` failed, Core will reset the SSL connection and restart a TLS handshake for the next `StartService` request.
```